xdsl/transforms/memref_stream_unnest_out_parameters.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass
+
+from xdsl.dialects import memref_stream
+from xdsl.dialects.builtin import (
+    AffineMapAttr,
+    ArrayAttr,
+    ModuleOp,
+)
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class UnnestOutParametersPattern(RewritePattern):
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, op: memref_stream.GenericOp, rewriter: PatternRewriter
+    ) -> None:
+        num_outputs = len(op.outputs)
+        if not num_outputs:
+            return
+
+        num_parallel = sum(
+            i == memref_stream.IteratorTypeAttr.parallel() for i in op.iterator_types
+        )
+        num_reduction = sum(
+            i == memref_stream.IteratorTypeAttr.reduction() for i in op.iterator_types
+        )
+        if num_parallel == len(op.iterator_types):
+            return
+
+        parallel_dims = (True,) * num_parallel + (False,) * num_reduction
```

Comment:
Don't forget to ensure this distribution in a verifier (can be done in subsequent PR).

Reply:
#2679: dialects: (memref_stream) add a verifier for order of iterator types in memref_stream.generic

Reply:
#2680